Error: build flag -mod=readonly only valid when using modules

Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version v1.32.2 built from (unknown, mod sum: "h1:CgIeFWTLJ3Nt1w/WU1RO351j/CjN6LIVjppbJfI9nMk=") on (unknown)
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
run:
  modules-download-mode: readonly
  skip-dirs:
    - .git
    - .github
    - .vscode
    - bin
    - dist
    - docs
    - man
    - tools

linters:
  disable-all: true
  enable:
    - bodyclose
    - deadcode
    - dogsled
    - dupl
    - errcheck
    - exhaustive
    - exportloopref
    - goconst
    - gofmt
    - goimports
    - golint
    - gosec
    - gosimple
    - govet
    - ineffassign
    - misspell
    - nolintlint
    - prealloc
    - staticcheck
    - structcheck
    - stylecheck
    - typecheck
    - unconvert
    - unparam
    - unused
    - varcheck
    - whitespace

linters-settings:
  gofmt:
    simplify: true
  goimports:
    local-prefixes: github.com/lukasmalkmus/xxx
  govet:
    check-shadowing: true
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.15.4 darwin/amd64
GO111MODULE="on"
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/lukasmalkmus/Library/Caches/go-build"
GOENV="/Users/lukasmalkmus/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/lukasmalkmus/Code/Go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/lukasmalkmus/Code/Go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/Cellar/go/1.15.4/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.15.4/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/lukasmalkmus/Code/Private/xxx/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/dy/s6dd70892l35f0444cwt96t40000gn/T/go-build856306592=/tmp/go-build -gno-record-gcc-switches -fno-common"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /Users/lukasmalkmus/Code/Private/xxx /Users/lukasmalkmus/Code/Private /Users/lukasmalkmus/Code /Users/lukasmalkmus /Users /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 27 linters: [bodyclose deadcode dogsled dupl errcheck exhaustive exportloopref goconst gofmt goimports golint gosec gosimple govet ineffassign misspell nolintlint prealloc staticcheck structcheck stylecheck typecheck unconvert unparam unused varcheck whitespace] 
INFO [loader] Go packages loading at mode 575 (deps|exports_file|imports|name|types_sizes|compiled_files|files) took 12.275422ms 
ERRO Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: build flag -mod=readonly only valid when using modules 
INFO Memory: 2 samples, avg is 72.1MB, max is 72.1MB 
INFO Execution took 25.264712ms  
```

</details>

Build details above should be sufficient enough. TL;DR: latest macOS Catalina, golangci-lint 1.32.2

The error only occurs when using `golang.org/x/tools v0.0.0-20201109182053-3db8fd265862`. When downgrading to `v0.0.0-20201022035929-9cf592e881e9` it works just fine.

Hey guys - we can confirm we are seeing this same error for v1.34.0 as well in our builds. We've been able to resolve for now by pinning to v1.33.2 👍🏻. Output from local lint here:

`golangci/golangci-lint info checking GitHub for latest tag
golangci/golangci-lint info found version: 1.34.0 for v1.34.0/linux/amd64
golangci/golangci-lint info installed /go/bin/golangci-lint
level=info msg="[config_reader] Config search paths: [./ /code / /root]"
level=info msg="[config_reader] Used config file .golangci.yml"
level=info msg="[lintersdb] Active 17 linters: [deadcode errcheck gocritic gocyclo golint gosimple govet ineffassign maligned misspell staticcheck structcheck typecheck unconvert unparam unused varcheck]"
level=info msg="[loader] Go packages loading at mode 575 (compiled_files|deps|files|types_sizes|exports_file|imports|name) took 7.036ms"
level=error msg="Running error: context loading failed: failed to load packages: failed to load with go/packages: err: exit status 1: stderr: build flag -mod=readonly only valid when using modules\n"
level=info msg="Memory: 2 samples, avg is 71.6MB, max is 71.6MB"
level=info msg="Execution took 18.4588ms"
make: *** [lint] Error 3`

--- a/.golangci.example.yml
+++ b/.golangci.example.yml
@@ -51,7 +51,7 @@ run:
   # If invoked with -mod=vendor, the go command assumes that the vendor
   # directory holds the correct copies of dependencies and ignores
   # the dependency descriptions in go.mod.
-  modules-download-mode: readonly|release|vendor
+  modules-download-mode: readonly|vendor|mod
 
   # Allow multiple parallel golangci-lint instances running.
   # If false (default) - golangci-lint acquires file lock on start.

--- a/go.mod
+++ b/go.mod
@@ -70,7 +70,7 @@ require (
 	github.com/valyala/quicktemplate v1.6.3
 	golang.org/x/sys v0.0.0-20201009025420-dfb3f7c4e634 // indirect
 	golang.org/x/text v0.3.4 // indirect
-	golang.org/x/tools v0.0.0-20210101214203-2dba1e4ea05c
+	golang.org/x/tools v0.0.0-20210105210202-9ed45478a130
 	gopkg.in/yaml.v2 v2.4.0
 	honnef.co/go/tools v0.0.1-2020.1.6
 	mvdan.cc/gofumpt v0.0.0-20201129102820-5c11c50e9475

--- a/pkg/lint/load.go
+++ b/pkg/lint/load.go
@@ -104,7 +104,7 @@ func (cl *ContextLoader) makeBuildFlags() ([]string, error) {
 	mod := cl.cfg.Run.ModulesDownloadMode
 	if mod != "" {
 		// go help modules
-		allowedMods := []string{"release", "readonly", "vendor"}
+		allowedMods := []string{"mod", "readonly", "vendor"}
 		var ok bool
 		for _, am := range allowedMods {
 			if am == mod {

